Core/src/org/sleuthkit/autopsy/actions/DeleteDataSourceAction.java

```diff
@@ -48,5 +51,13 @@ public void actionPerformed(ActionEvent event) {
             logger.log(Level.WARNING, "Error Deleting Data source " + selectedDataSource, e);
         }
     }
-
+    private static void deleteDataSource(Long dataSourceId) throws KeywordSearchServiceException {
+        try {
+            KeywordSearchService kwsService = Lookup.getDefault().lookup(KeywordSearchService.class);
+            kwsService.deleteDataSource(dataSourceId);
+        } catch (KeywordSearchServiceException e) {
+            logger.log(Level.WARNING, "KWS Error", e);
+        }
```

Comment:
We may have already added this in a later PR, in which case this comment can be ignored...

If the action fails, we should give the user some immediate feedback with a popup from MessageUtil, in addition to logging the error. To this end, the static method should propagate the KeywordSearchServiceException (as it erroneously says it does) so that actionPerformed can take responsibility for comprehensive error handling. I see this as a micro application of the guideline from our coding standard that says that lower level code should leave error handling policy to top level code.

Reply:
This is in a later PR 2415 I have confirmed it.

Reply:
Cool, I'm content to wait for the user notification.

However, we are still catching the KeywordSearchServiceException in DeleteDataSourceAction.deleteDataSource instead of propagating it to performAction as we should and the exception specification for DeleteDataSourceAction.deleteDataSource states. 

Feel free to refactor the code in DeleteDataSourceAction.deleteDataSource into DeleteDataSourceAction.performAction and delete the DeleteDataSourceAction.deleteDataSource method. Or propagate the exception by removing the try-catch in  DeleteDataSourceAction.deleteDataSource.